cohttp-lwt-unix-tls/src/net.ml

```diff
@@ -0,0 +1,60 @@
+(*{{{ Copyright (c) 2012 Anil Madhavapeddy <anil@recoil.org>
+ *
+ * Permission to use, copy, modify, and distribute this software for any
+ * purpose with or without fee is hereby granted, provided that the above
+ * copyright notice and this permission notice appear in all copies.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+ * WITH REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF
+ * MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+ * ANY SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+ * WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER IN AN
+ * ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+ * OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+ *
+  }}}*)
+
+(* Miscellaneous net-helpers used by Cohttp. Ideally, these will disappear
+ * into some connection-management framework such as andrenth/release *)
+
+open Lwt.Infix
+
+module IO = Cohttp_lwt_unix.IO
+
+type resolvers = Conduit.resolvers
+
+let () = Mirage_crypto_rng_unix.initialize ()
+
+let authenticator ~host:_ _ = Ok None
+let config = Tls.Config.client ~authenticator ()
+
+let empty =
+  Conduit_lwt.empty
+  |> Conduit_lwt.add Conduit_lwt.TCP.protocol
+    (Conduit_lwt.TCP.resolve ~port:80)
+  |> Conduit_lwt.add ~priority:10 Conduit_lwt_tls.TCP.protocol
+    (Conduit_lwt_tls.TCP.resolve ~port:443 ~config)
+
+let failwith fmt = Format.kasprintf (fun err -> Lwt.fail (Failure err)) fmt
+
+let connect_uri ?host:(default= "localhost") ~resolvers uri =
```

Comment:
`?host` doesn't actually seem to be used here. What is it for?

Reply:
It is used by `Uri.host_with_default ~default` where we want to extract the `domain-name` from the `Uri.t`. `Uri.t` is super-versatile in this case and the cast can lead to an error when you give something like: `https://0.0.0.0/` - throught, I started to discuss about this specific point [here](https://github.com/mirage/ocaml-conduit/pull/311#issuecomment-694970864).